Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1220,40 +1222,61 @@ namespace Multiplayer
+            if (multiplayer->StartHosting(sv_port, sv_isDedicated))
+            {
+                AZLOG_ERROR("Failed to start listening on any allocated port");
+            }
+        }
+        else if (const auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+        {
+            settingsRegistry->Set(DeferredHostKey, true);
         }
     }
     AZ_CONSOLEFREEFUNC(host, AZ::ConsoleFunctorFlags::DontReplicate, "Opens a multiplayer connection as a host for other clients to connect to");
 
     void connect([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
     {
-        if (arguments.size() < 1)
-        {
-            const AZ::CVarFixedString remoteAddress = cl_serveraddr;
-            AZ::Interface<IMultiplayer>::Get()->Connect(remoteAddress.c_str(), cl_serverport);
-        }
-        else
+        if (IMultiplayer* multiplayer = AZ::Interface<IMultiplayer>::Get())
         {
-            AZ::CVarFixedString remoteAddress{ arguments.front() };
-            const AZStd::size_t portSeparator = remoteAddress.find_first_of(':');
-            if (portSeparator == AZStd::string::npos)
+            if (arguments.empty())
             {
-                AZ::Interface<IMultiplayer>::Get()->Connect(remoteAddress.c_str(), cl_serverport);
+                const AZ::CVarFixedString remoteAddress = cl_serveraddr;
+                multiplayer->Connect(remoteAddress.c_str(), cl_serverport);
             }
             else
             {
-                char* mutableAddress = remoteAddress.data();
-                mutableAddress[portSeparator] = '\0';
-                const char* addressStr = mutableAddress;
-                const char* portStr = &(mutableAddress[portSeparator + 1]);
-                int32_t portNumber = atol(portStr);
-                AZ::Interface<IMultiplayer>::Get()->Connect(addressStr, static_cast<uint16_t>(portNumber));
+                AZ::CVarFixedString remoteAddress{ arguments.front() };
+                const AZStd::size_t portSeparator = remoteAddress.find_first_of(':');
+                if (portSeparator == AZStd::string::npos)
+                {
+                    multiplayer->Connect(remoteAddress.c_str(), cl_serverport);
+                }
+                else
+                {
+                    char* mutableAddress = remoteAddress.data();
+                    mutableAddress[portSeparator] = '\0';
+                    const char* addressStr = mutableAddress;
+                    const char* portStr = &(mutableAddress[portSeparator + 1]);
+                    int32_t portNumber = atol(portStr);
+                    multiplayer->Connect(addressStr, static_cast<uint16_t>(portNumber));
+                }
             }
         }
+        else if (const auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+        {
+            settingsRegistry->Set(DeferredConnectKey, true);
```

Comment:
Maybe add a comment here describing why this is "connect logic" is doing - that we are setting up for a deferred connection later because Multiplayer is not ready to handle connection currently?

Reply:
Added.